src/box/index.cc

```diff
@@ -936,6 +936,45 @@ generic_index_count(struct index *index, enum iterator_type type,
 	return count;
 }
 
+#if defined(ENABLE_READ_VIEW)
+
+ssize_t
+generic_index_read_view_count(struct index_read_view *rv,
+			      enum iterator_type type, const char *key,
+			      uint32_t part_count)
+{
+	struct index_read_view_iterator it;
+	if (index_read_view_create_iterator(rv, type, key,
+					    part_count, &it) != 0)
+		return -1;
+	int rc = 0;
+	size_t count = 0;
+	struct tuple *tuple = NULL;
+	while ((rc = box_index_read_view_iterator_next(&it, &tuple)) == 0 &&
```

Comment:
Same comment as to `generic_index_read_view_create_iterator_with_offset` - no need to use `box_index_read_view_iterator_next` here, `next_raw` should be enough.

Also, I'd merge this commit with the one introducing `create_iterator_with_offset`.

Reply:
Done.

Regarding merging the commits. If you have no objections, I keep these two: it seems to me like they should be separated since they introduce unrelated read view methods.